packages/@aws-cdk-testing/framework-integ/test/aws-codebuild/test/integ.docker-registry-build-images.ts

```diff
@@ -0,0 +1,125 @@
+import * as integ from '@aws-cdk/integ-tests-alpha';
+import * as assert from 'assert';
+import * as cdk from 'aws-cdk-lib';
+import * as codebuild from 'aws-cdk-lib/aws-codebuild';
+import * as secretsmanager from 'aws-cdk-lib/aws-secretsmanager';
+import { Construct } from 'constructs';
+
+const app = new cdk.App();
+
+const BuildImageTestCases = {
+  LinuxBuildImage: {
+    computeType: codebuild.ComputeType.SMALL,
+    generateBuildImage: (secretsManagerCredentials: secretsmanager.ISecret) =>
+      codebuild.LinuxBuildImage.fromDockerRegistry('alpine', {
+        secretsManagerCredentials,
+      }),
+  },
+  LinuxArmBuildImage: {
+    computeType: codebuild.ComputeType.SMALL,
+    generateBuildImage: (secretsManagerCredentials: secretsmanager.ISecret) =>
+      codebuild.LinuxArmBuildImage.fromDockerRegistry('alpine', {
+        secretsManagerCredentials,
+      }),
+  },
+  // FIXME remove LinuxArmLambdaBuildImage.fromDockerRegistry
+  // LinuxLambdaBuildImage: { computeType: codebuild.ComputeType.LAMBDA_1GB },
+  // LinuxArmLambdaBuildImage: { computeType: codebuild.ComputeType.LAMBDA_1GB },
+  WindowsBuildImage: {
+    computeType: codebuild.ComputeType.MEDIUM,
+    generateBuildImage: (secretsManagerCredentials: secretsmanager.ISecret) =>
+      codebuild.WindowsBuildImage.fromDockerRegistry(
+        'mcr.microsoft.com/windows/servercore:ltsc2019',
+        { secretsManagerCredentials },
+        codebuild.WindowsImageType.SERVER_2019,
+      ),
+  },
+};
+const buildImageClassNames = Object.keys(
+  BuildImageTestCases,
+) as (keyof typeof BuildImageTestCases)[];
+
+interface DockerRegistryBuildImageTestStackProps extends cdk.StackProps {
+  buildImageClassName: keyof typeof BuildImageTestCases;
+  completeSecretArn: string;
+  computeType: codebuild.ComputeType;
+}
+
+const secretStack = new cdk.Stack(app, 'SecretStack');
+const { secretFullArn } = new secretsmanager.Secret(secretStack, 'MySecret', {
+  secretStringValue: cdk.SecretValue.unsafePlainText('my-secret-auth'),
+});
```

Comment:
I would have liked to integ test `secretsManagerCredentials`, but unfortunately its value is being checked even if the images being pulled are public, which makes sense, anonymous pulls get throttled pretty aggressively. Not a huge deal, but I don't particularly feel like creating an account, publishing its credentials here, and creating private images just to test this 😅